schema/spacewalk/common/views/userAccessTable.sql

```diff
@@ -0,0 +1,33 @@
+--
+-- Copyright (c) 2025 SUSE LLC
+--
+-- This software is licensed to you under the GNU General Public License,
+-- version 2 (GPLv2). There is NO WARRANTY for this software, express or
+-- implied, including the implied warranties of MERCHANTABILITY or FITNESS
+-- FOR A PARTICULAR PURPOSE. You should have received a copy of GPLv2
+-- along with this software; if not, see
+-- http://www.gnu.org/licenses/old-licenses/gpl-2.0.txt.
+--
+
+-- userAccessTable view
+-- User access rules on endpoints either permitted directly or through an access group
+CREATE VIEW access.userAccessTable AS
+WITH endpoints AS (
```

Comment:
I'm not sure if we need this initial query. We should test it in terms of performance when we have all the endpoints and see if it has any performance impact.

Reply:
This one used to be for my own convenience, but I changed this view completely with the latest commit. This view is now queried from the Java filters and is an integral part of the feature, so its performance is critical and I optimized it with proper indexes and planning.

So far I tested it with what data I have, and it's running in microseconds (with about half of the predicted endpoint data).